### PR TITLE
Make reuse of sql test code explicit

### LINKED ISSUE #45866: [CI] JdbcSqlSpecIT and JdbcCsvSpecIT initializationError
My [PR build failed](https://gradle-enterprise.elastic.co/s/dg7sr4qdluup2/tests/failed) with initializationError in above tests due to a missing directory. This seems to have happened once before on August 14th. I can reproduce this locally too.

The error is:
```
org.elasticsearch.xpack.sql.qa.security.JdbcCsvSpecIT > initializationError FAILED
--
java.nio.file.NoSuchFileException: /var/lib/jenkins/workspace/elastic+elasticsearch+pull-request-2/x-pack/plugin/sql/qa/security/without-ssl/build/classes/java/test
```

I can reproduct this locally by doing:
```
./gradlew :x-pack:plugin:sql:qa:security:without-ssl:clean
```
and then
```
./gradlew :x-pack:plugin:sql:qa:security:without-ssl:integTestRunner
```
fails. The `test` directory referenced above does not exist.

 If I instead invoke the `check` task, it works OK and the directory is created.

### PR DESCRIPTION
**Pull request by rjernst**
The sql project uses a common set of security tests, which are run in
subprojects. Currently these are shared through a shared directory, but
this is not setup correctly to ensure it is built before tests run. This
commit changes the test classes to be an artifact of the sql/qa/security
project and makes the test runner use the built artifact (a directory of
classes) for tests.

closes #45866

### CODE REVIEW

**Review by alpar-t — APPROVED** [review]
LGTM

**Comment by rjernst** [comment]
@elasticmachine run elasticsearch-ci/1

**Comment by rjernst** [comment]
@elasticmachine run elasticsearch-ci/1

File: x-pack/plugin/sql/qa/security/build.gradle [hunk 1]
```diff
@@ -42,10 +43,22 @@ subprojects {
     user username: "test_admin", password: "x-pack-test-password"
   }
 
+  File testArtifactsDir = project.file('build/testArtifacts')
```
Comment (mark-vieira) [inline x-pack/plugin/sql/qa/security/build.gradle]: Let's use `$buildDir` instead of hard-coding `build`.

**Review by mark-vieira — APPROVED** [review]
One minor comment, otherwise changes LGTM 👍

**Pull request merged**